tm2/pkg/os/tempfile.go

```diff
@@ -107,14 +107,19 @@ func WriteFileAtomic(filename string, data []byte, perm os.FileMode) (err error)
 		}
 		break
 	}
+
+	// Clean up in any case.
+	defer func() {
+		if f != nil {
```

Comment:
f cannot be nil at this point

Reply:
It could be if `for ; i < atomicWriteFileMaxNumWriteAttempts; i++ {` ended without getting back a valid file.

Reply:
No, because we check for `err != nil`, so at this stage `err == nil` so there must be a file.

Reply:
True true, let me update it and thank you for the discourse.

Reply:
Done, thanks!